src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -139,24 +140,34 @@ public virtual Transmission Dequeue()
                         return transmission;
                     }
                 }
-                catch (UnauthorizedAccessException)
+                catch (UnauthorizedAccessException uae)
                 {
                     if (file == null)
                     {
-                        return null; // because the process does not have permission to access the folder
+                        TelemetryChannelEventSource.Log.TransmissionStorageDequeueUnauthorizedAccessException(this.folder.Name, uae.ToString());
+                        return null; // Because the process does not have permission to access the folder.
                     }
 
-                    if (lastInaccessibleFileName != file.Name)
+                    string name = file.Name;
+                    if (this.badFiles.TryAdd(name, null))
+                    {
+                        TelemetryChannelEventSource.Log.TransmissionStorageInaccessibleFile(name);
+                    }
+                    else
                     {
-                        lastInaccessibleFileName = file.Name;
-                        continue; // because another thread is loading this file right now
+                        // The same file has been inaccessible more than once because the process does not have permission to modify this file.
+                        TelemetryChannelEventSource.Log.TransmissionStorageUnexpectedRetryOfBadFile(name);
                     }
 
-                    throw; // because the process does not have permission to modify this file
+                    Thread.Sleep(random.Next(1, 100)); // Sleep for random time of 1 to 100 milliseconds to try to avoid future timing conflicts.
                 }
-                catch (IOException)
+                catch (IOException ioe)
                 {
-                    continue; // because another thread already loaded this file
+                    // This exception can happen when one thread runs out of files to process and reloads the list while another
+                    // thread is still processing a file and has not deleted it yet thus allowing it to get in the list again.
+                    TelemetryChannelEventSource.Log.TransmissionStorageDequeueIOError(file.Name, ioe.ToString());
```

Comment:
I've put a little sleep with a comment before the file list is loaded which should mitigate this exception ever happening (at least for the scenario I was able to repro in testing).